tools/build_pak_files.py

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+import argparse
+import glob
+import os
+import posixpath
+import spear
+import subprocess
+import sys
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--unreal_engine_dir", required=True)
+    parser.add_argument("--contents_dir", required=True)
+    parser.add_argument("--platforms", nargs="*", required=True)
+    parser.add_argument("--output_dir", required=True)
+    args = parser.parse_args()
+    
+    assert os.path.exists(args.unreal_engine_dir)
+    if sys.platform == "win32":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Win64", "UE4Editor.exe"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Win64", "UnrealPak.exe"))
+    elif sys.platform == "darwin":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Mac", "UE4Editor"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Mac", "UnrealPak"))
+    elif sys.platform == "linux":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Linux", "UE4Editor.sh"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_engine_dir, "Binaries", "Linux", "UnrealPak.sh"))
+
+    unreal_project_dir         = os.path.realpath(os.path.join(os.path.dirname(__file__), "..", "cpp", "unreal_projects", "SpearSim"))
+    uproject                   = os.path.join(unreal_project_dir, "SpearSim.uproject")
+    unreal_project_content_dir = os.path.join(unreal_project_dir, "Content")
+
+    assert os.path.exists(args.contents_dir)
+    shared_dir = os.path.realpath(os.path.join(args.contents_dir, "Shared"))
+    scenes_dir = os.path.realpath(os.path.join(args.contents_dir, "Scenes"))
+
+    assert os.path.exists(shared_dir)
+    unreal_project_content_shared_dir = os.path.join(unreal_project_content_dir, "Shared")
+    if not spear.path_exists(unreal_project_content_shared_dir):
+        print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_shared_dir} -> {shared_dir}")
+        os.symlink(shared_dir, unreal_project_content_shared_dir)
+
+    assert os.path.exists(scenes_dir)
+    scene_content_dirs = [ os.path.realpath(os.path.join(scenes_dir, x)) for x in os.listdir(scenes_dir) if x.startswith("kujiale") ]
```

Comment:
I don't think this script should know about specific "scene groups" (e.g., all the `kujiale_*` scenes). If anything, the user should be able to provide an optional scene name or a wild-card string that the scene IDs are matched against. There are examples of how to do this cleanly in the hypersim codebase, search for `fnmatch`.

Reply:
Added a wild-card string based scene selection using `fnmatch`. Using `fnmatch` is pretty convenient and clean, thanks for the suggestion.